### PR TITLE
fix: parsing of browserslist

### LINKED ISSUE #11876: In Webpack 5 a target containing IE 11 doesn't build the runtime in ES5
# Bug report

**What is the current behavior?**

In Webpack 5 a [`target`](https://webpack.js.org/configuration/target/) containing Internet Explorer 11, e.g. `target: 'browserslist:ie 11',`, doesn't build the runtime in ES5.

**If the current behavior is a bug, please provide the steps to reproduce.**

1. Have a target which includes IE 11, e.g. `target: 'browserslist:ie 11'`
2. Run webpack
3. Try to load your application in IE 11
4. See errors in the console, e.g.
   [SCRIPT1010](https://docs.microsoft.com/de-de/scripting/javascript/misc/expected-identifier-javascript): Expected identifier
    ```
   /******/ 		// install a JSONP callback for chunk loading
   /******/ 		var webpackJsonpCallback = function(data) {
   /******/ 			var [chunkIds, moreModules, runtime, executeModules] = data; // <- this line
   ```

**What is the expected behavior?**

Runtime is ES5 if IE 11 is present, like `target: ['web', 'es5']` mentioned in https://webpack.js.org/migrate/5/#turn-off-es2015-syntax-in-runtime-code-if-necessary.

**Other relevant information:**
webpack version: 5.3.2
Node.js version: 14.7.0
Operating System: Win10 2004
Additional tools: Powershell

Related to #11814 or #11753 maybe?

The SCRIPT1010 error above exist when running `mode: 'development',`, in `mode: 'production',` I get [SCRIPT1003](https://docs.microsoft.com/de-de/scripting/javascript/misc/expected-colon): Expected ':' in one of our bundles, so it's probably not even limited to the runtime?

As hinted, using `target: ['web', 'es5']` works fine in both cases. `'browserslist:IE >= 10'` like in #11753 works as well. But both these workarounds aren't the same as targeting IE 11.

### PR DESCRIPTION
**Pull request by evilebottnawi**
**What kind of change does this PR introduce?**

fixes #11876, found a lot of edge cases with tests, I think we should switch from browserslist supports feature because a lot problems, babel also keep browsers in own code and use only syntax of browserslist

WIP on more fixes

**Did you add tests for your changes?**

Yes

**Does this PR introduce a breaking change?**

No

**What needs to be documented once your changes are merged?**

No

### CODE REVIEW

File: lib/config/browserslistTargetHandler.js [hunk 1]
```diff
@@ -75,80 +75,205 @@ const load = (input, context) => {
...
+			samsung: [8, 0],
+			android: 61,
+			and_qq: [10, 4],
+			// baidu: Not supported
+			// and_uc: Not supported
+			// kaios: Not supported
+			// Since Node.js 13.14.0 no warning about usage, but it was added 8.5.0 with some limitations and it was improved in 12.0.0 and 13.2.0
+			node: [13, 14]
+		}),
+		//	browserslistChecker("es6-module") && rawChecker({ node: [12, 17] }),
```
Comment (sokra) [inline lib/config/browserslistTargetHandler.js:254]:
```suggestion
```

**Review by sokra — CHANGES_REQUESTED** [review]

**Review by sokra — APPROVED** [review]
I don't want to review all this numbers, but I trust you that that's fine. If you think this is done, I'll merge it.

**Comment by evilebottnawi** [comment]
Yep, let's merge, I have given comments in many places and we can always fix it in future

**Comment by vankop** [comment]
probably webpack needs some e2e browser testing =)

**Comment by evilebottnawi** [comment]
@vankop yep, it will be great :smile:

**Comment by sokra** [comment]
Thanks
